src/components/editor/MonacoEditor/index.tsx

```diff
@@ -8,35 +15,40 @@ import {
     useEditorStore_currentCatalog,
     useEditorStore_serverUpdate,
     useEditorStore_setStatus,
-    useEditorStore_status,
+    useEditorStore_statuses,
 } from 'components/editor/Store/hooks';
 import { EditorStatus } from 'components/editor/Store/types';
 import { debounce } from 'lodash';
 import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
-import { useCallback, useMemo, useRef, useState } from 'react';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { logRocketConsole } from 'services/logrocket';
 import { stringifyJSON } from 'services/stringify';
+import { Entity } from 'types';
 import {
     DEFAULT_HEIGHT,
     DEFAULT_TOOLBAR_HEIGHT,
     ICON_SIZE,
 } from 'utils/editor-utils';
 import { AllowedScopes } from './types';
 
-type onChange = (
+type EditorChangeHandler = (
     newVal: any,
     path: string,
-    specType: string,
-    scope?: AllowedScopes
+    specType: Entity,
+    scope?: AllowedScopes | string
```

Comment:
Either the AllowedScopes or the string type needs removed. Allowing any string defeats the purpose of having a known list of scopes.

Reply:
This was something I planned to revisit before merging. The editor has gone through many iterations and I decided to make a final call once the component was in a stable state. Leaning towards removing `AllowedScopes` altogether.